dbms/src/Storages/DeltaMerge/ReadThread/CPU.cpp

```diff
@@ -0,0 +1,68 @@
+#include <Storages/DeltaMerge/ReadThread/CPU.h>
+#include <Poco/File.h>
+#include <Poco/DirectoryIterator.h>
+#include <string>
+
+namespace DB::DM
+{
+// In Linux a numa node is represented by a device directory, such as '/sys/devices/system/node/node0', ''/sys/devices/system/node/node01'.
```

Comment:
I wonder whether it is a common case on all platforms, other than linux. We better do some protection, like rollback strategy on other platforms.

Reply:
some examples are at: https://github.com/microsoft/mimalloc/blob/master/src/os.c